app/controllers/cities_controller.rb

```diff
@@ -36,49 +33,50 @@ def a_in_b_as_c?(a, b, c) # a in b as c
       return false
     end
     
-    # Making a cookie to store all data of cities being searched.
-    def city_data
-      if params[:geo]
-        latlng = params[:geo]
-        loc_key = City.get_loc_key(latlng["lat"], latlng["lng"], params[:name])
-        city = City.find_by(location_key: loc_key)
+    # Helper method to reduce code complexity
+    def update_recent_cities(user, rec_cities)
+      if rec_cities.length > 5
+        rec_cities = rec_cities[rec_cities.length - 5, rec_cities.length - 1]
       end
-      city.update_city_data
-
+      if user.provider.nil?
+        user.update_attributes(recent_cities: rec_cities)
+        user.save(:validate => false)
+      else
+        user.update_attributes!(recent_cities: rec_cities)
+      end
+      user.recent_cities.reverse
```

Comment:
I think `update_attributes` saves the user record so i don't think `user.save(:validate => false)` will help avoid the error.
Lines 41-46 could be replaced with `user.recent_cities = rec_cities` and  `user.save(:validate => false)`

Reply:
Noted! Will implement in next patch